### ISSUE TITLE
S3453: Consider target-typed new expressions from C# 9.0

### ISSUE THREAD

**Issue by anreton** [SonarSource/sonar-dotnet#4529]
### Description
[S3453](https://rules.sonarsource.com/csharp/RSPEC-3453)

### Repro steps

```csharp
public class Foo
{
    public static readonly Foo Instance = new();

    private Foo() {}
}
```

### Expected behavior

No false diagnostic for S3453.

### Actual behavior

Get false diagnostic for S3453.

### Related information

- Microsoft Visual Studio Community 2019 Version 16.10.0
- SonarAnalyzer.CSharp 8.23.0.32424 via Nuget
- https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/target-typed-new

**Comment by pavel-mikula-sonarsource** [comment]
Hi @anreton,

We've released https://github.com/SonarSource/sonar-dotnet/releases/tag/8.24.0.32949 this Monday with support for C# 9 target typed new support that also addresses S3453.
